From bf8d9ae1843a8f5739a782642c2699b472bd3768 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@debian.org>
Date: Thu, 30 Apr 2015 22:38:53 +0200
Subject: [PATCH] dh_makeshlibs: Do not inject multiarch-support in
 misc:Pre-Depends anymore

This transition has long been completed. The version introducing this
change was 8.1.3 shipped with squeeze, the release that included the
relevant changes in glibc and the dynamic linker.
---
 dh_makeshlibs | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/dh_makeshlibs b/dh_makeshlibs
index c315641..d32e771 100755
--- a/dh_makeshlibs
+++ b/dh_makeshlibs
@@ -21,11 +21,6 @@ libraries, and generates a shlibs file for the libraries it finds.
 It also adds a call to ldconfig in the F<postinst> and F<postrm> scripts (in
 v3 mode and above only) to any packages in which it finds shared libraries.
 
-Packages that support multiarch are detected, and
-a Pre-Dependency on multiarch-support is set in ${misc:Pre-Depends} ;
-you should make sure to put that token into an appropriate place in your
-debian/control file for packages supporting multiarch.
-
 =head1 FILES
 
 =over 4
@@ -136,7 +131,6 @@ init(options => {
 });
 
 my $objdump=cross_command("objdump");
-my $multiarch=dpkg_architecture_value("DEB_HOST_MULTIARCH");
 
 my $ok=1;
 
@@ -147,7 +141,6 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 
 	my %seen;
 	my $need_ldconfig = 0;
-	my $is_multiarch = 0;
 	my $shlibs_file = pkgfile($package, 'shlibs');
 
 	doit("rm", "-f", "$tmp/DEBIAN/shlibs");
@@ -167,9 +160,6 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 	while (<FIND>) {
 		my ($library, $major);
 		push @lib_files, $_;
-		if (defined $multiarch && $multiarch ne '' && m,/$multiarch/,) {
-			$is_multiarch=1;
-		}
 		my $ret=`$objdump -p $_`;
 		if ($ret=~m/\s+SONAME\s+(.+)\.so\.(.+)/) {
 			# proper soname format
@@ -281,9 +271,6 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 			doit("rm", "-f", "$tmp/DEBIAN/symbols");
 		}
 	}
-	if ($is_multiarch) {
-		addsubstvar($package, "misc:Pre-Depends", "multiarch-support");
-	}
 }
 
 unless ($ok) {
-- 
2.2.1.209.g41e5f3a

